framework/src/modules/chain/dpos/delegates_list.js

```diff
@@ -128,13 +128,18 @@ class DelegatesList extends EventEmitter {
 	 * Validates if block was forged by correct delegate
 	 *
 	 * @param {Object} block
+	 * @param {Number} roundOffset - use delegate list generated at the end of `roundOffset` before the current round
 	 * @return {Boolean} - `true`
 	 * @throw {Error} Failed to verify slot
 	 */
-	async verifyBlockForger(block) {
+	async verifyBlockForger(block, roundOffset = 0) {
 		const currentSlot = this.slots.getSlotNumber(block.timestamp);
-		const round = this.slots.calcRound(block.height);
-		const delegateList = await this.getForgerPublicKeysForRound(round);
+		const currentRound = this.slots.calcRound(block.height);
+
+		// Delegate list is generated from round 1 hence `roundToVerify` can't be less than 1
+		const roundToVerify = Math.max(currentRound - roundOffset, 1);
```

Comment:
wouldn't it be better if we throw an exception? If we provide roundOffset = 3, seems like the first 2 rounds will be calculated as round 1?

Reply:
That's the idea. From the LIP:

> Note that for a new blockchain, this delay of 2 rounds can be achieved by letting the active delegates of the first three rounds be the delegates defined in the genesis block.